### DISCUSSION TITLE
Unable to install breeze

### DISCUSSION THREAD

**Discussion by khtee** [apache/airflow#35230] [General]
I'm trying to install Breeze for running local test. I hit the following error on Ubuntu 20.04.

Command:
```
pipx install -e ./dev/breeze/
```

Error Message:
```
/usr/bin/pipx:6: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import load_entry_point
Could not find package ./dev/breeze/. Is the name correct?
```

**Comment by potiuk** [comment]
You are liklyt not in the right folder. the `./dev/breeze` is a path relative from sources of Airflow.

**Comment by khtee** [comment]
I'm at the root path of the cloned repo.

```bash
user@ubuntu:/tmp$ git clone git@github.com:apache/airflow.git && cd airflow
Cloning into 'airflow'...
remote: Enumerating objects: 363384, done.
remote: Counting objects: 100% (70177/70177), done.
remote: Compressing objects: 100% (5192/5192), done.
remote: Total 363384 (delta 67872), reused 65048 (delta 64985), pack-reused 293207
Receiving objects: 100% (363384/363384), 240.03 MiB | 7.54 MiB/s, done.
Resolving deltas: 100% (283699/283699), done.

user@ubuntu:/tmp/airflow$ pipx install -e ./dev/breeze/
/usr/bin/pipx:6: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  from pkg_resources import load_entry_point
Could not find package ./dev/breeze/. Is the name correct?

user@ubuntu:/tmp/airflow$           
```

**Reply by potiuk** [reply]
> I'm at the root path of the cloned repo.

Does the pth exist? Is there a proper breeze package in your checkout? Is pipx able to install other packages? Maybe you should look at those things first? Ir maybe you have some permissions settings?  Or maybe your `pipx` is somewhat broken or old

I suggest you to install latest version of `pipx` - I have no idea what version you have, also some distributions install some versions of Python, pip, possibly pipx that are limited in functionality and somewhat broken. 

So everyone who is a `Python developer` and is serious in developing anything in Python and not only a user of system libraries using python as runtime is recommend (by pip maintainers and generally Python Packaging group) to do not use those, but manage your Python on your own in local virtualenvs. 

I personally recommend  Pyenv to manage your Python, Pip and Pipx on your distribution. If you do not know these tools - I strongly recommend to get familiar with them if you want to contribute to any Python project.

Generally look at `pipx` - familiarize yourself with what it does - it installs Python packages. You might also take a look at BREEZE.rst documentatiom for some troubleshooting steps as well.

**Reply by potiuk** [reply]
(And yes, the Deprecation warning you see suggest that your basic Python toolchain has problems and going the `pyenv` route to install everything that you need is the best way to do Python software development)

**Comment by potiuk** [comment]
Please do report it back when you have latest Python versions (3.8 is minimum supported, but I recommend 3.11) + latest pip + latest pipx managed by Pyenv, and you checked all the permission problems.

Also remember that for breeze you need docker and doker-compose also in rather new versions, so you will need to install those as well (but breeze will tell you if you will have too old versions of those once you manage to bootstrap it).

**Comment by khtee** [comment]
The path is valid. Below are the version for Python, pip and pipx,

```
Python 3.8.10
pip 23.3.1 
pipx 0.12.3.1
```

Pipx has no problem install other packages.

**Comment by khtee** [comment]
I think the problem might be from the pipx install using `apt-get install pipx -y`

I switched to using the following method I can install breeze successfully.
```bash
$ python3 -m pip install pipx
$ python3 -m pipx ensurepath
$ python3 -m pipx install ./dev/breeze
```

**Reply by potiuk** [reply]
Yeah. That's what I expected
